indexing-service/src/main/java/org/apache/druid/indexing/common/task/KillUnusedSegmentsTask.java

```diff
@@ -346,4 +306,32 @@ public LookupLoadingSpec getLookupLoadingSpec()
   {
     return LookupLoadingSpec.NONE;
   }
+
+  @Override
+  public boolean isReady(TaskActionClient taskActionClient) throws Exception
+  {
+    final boolean useConcurrentLocks = Boolean.TRUE.equals(
+        getContextValue(
+            Tasks.USE_CONCURRENT_LOCKS,
+            Tasks.DEFAULT_USE_CONCURRENT_LOCKS
+        )
+    );
+
+    TaskLockType defaultLockType = useConcurrentLocks ? TaskLockType.REPLACE : TaskLockType.EXCLUSIVE;
+
+    TaskLockType actualLockType = getContextValue(Tasks.TASK_LOCK_TYPE, defaultLockType);
```

Comment:
IIUC, this code means that if a user has specified both `useConcurrentLocks` and `taskLockType` in the task context, the latter will be preferred.
But in `AbstractBatchIndexTask.determineLockType()`, `useConcurrentLocks` is given preference. We should stick to the existing preference order.

Reply:
thanks! updated code to prefer useConcurrentLocks lock (replace)